src/kOS.Safe/Execution/CPU.cs

```diff
@@ -413,11 +415,13 @@ public void AssertValidDelegateCall(IUserDelegate userDelegate)
         {
             if (userDelegate.ProgContext != currentContext)
             {
-                throw new KOSInvalidDelegateContextException(
-                    (currentContext == contexts[0] ? "the interpreter" : "a program"),
-                    (currentContext == contexts[0] ? "a program" : "the interpreter")
-                    );
-            }
+                string currentContextName = (currentContext == contexts[0] ? "the interpreter" : "a program");
+                string delegateContextName = (
+                    userDelegate.ProgContext == contexts[0] ? currentContextName = "the interpreter" : (
+                        currentContext == contexts[0] ? "a program" : "a different program from a previous run" ) );
+
+                throw new KOSInvalidDelegateContextException(currentContextName, delegateContextName);
+           }
```

Comment:
That's kind of what I expected, however I'm not sure that's what the chain is doing.  Maybe I'm reading it wrong.  I read this portion:
```
string delegateContextName = (
                    userDelegate.ProgContext == contexts[0] ? currentContextName = "the interpreter" : \\ the rest of the chain
```
_(specifically the single `=` after currentContextName)_

to mean
```
if (userDelegate.ProgContext == contexts[0])
{
    delegateContextName = currentContextName = "the interpreter";
}
else // continue down the chain
```
Am I reading it wrong, or is it behaving differently than intended?